django/contrib/postgres/search.py

```diff
@@ -381,3 +396,91 @@ class TrigramWordSimilarity(TrigramWordBase):
 
 class TrigramStrictWordSimilarity(TrigramWordBase):
     function = "STRICT_WORD_SIMILARITY"
+
+
+class LexemeCombinable:
+    BITAND = "&"
+    BITOR = "|"
+
+    def _combine(self, other, connector, reversed, node=None):
+        if not isinstance(other, LexemeCombinable):
+            raise TypeError(
+                "A Lexeme can only be combined with another Lexeme, "
+                f"got {type(other)}."
+            )
```

Comment:
This is untested.

(Also under the strictest interpretation of the style guideline about computations in f-strings, you might be asked to prefer only attribute access, e.g. `other.__class__.__name__`.)